lib/MonadicRewrite.thy

```diff
@@ -763,7 +766,7 @@ method monadic_rewrite_l_method methods action finalise =
 
 (* Step over RHS until action applies, then finalise. *)
 method monadic_rewrite_r_method methods action finalise =
-  monadic_rewrite_single_pass \<open>wp_pre, rule monadic_rewrite_trans[rotated]\<close>
+  monadic_rewrite_single_pass \<open>monadic_rewrite_pre, rule monadic_rewrite_trans[rotated]\<close>
```

Comment:
wait, how is this gonna work if you haven't added any theorems to `monadic_rewrite_pre`?

Reply:
right, it's not going to work :-). I forgot to add the thm after declaring the theorem set

Reply:
Ok, added now.